### ISSUE TITLE
Operator fails to install on K8s >= v1.18.x

### ISSUE THREAD

**Issue by jthomperoo** [jthomperoo/custom-pod-autoscaler-operator#73]
**Describe the bug**
The CPAO fails to install on K8s >= `v1.18.x`

**To Reproduce**
1. Install the operator on a K8s cluster `v1.18.x`:

```bash
VERSION=v1.0.2
HELM_CHART=custom-pod-autoscaler-operator
helm install ${HELM_CHART} https://github.com/jthomperoo/custom-pod-autoscaler-operator/releases/download/${VERSION}/custom-pod-autoscaler-operator-${VERSION}.tgz
```

2. Error reported, install fails:

```
Error: CustomResourceDefinition.apiextensions.k8s.io "custompodautoscalers.custompodautoscaler.com" is invalid: [spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property, spec.validation.openAPIV3Schema.properties[spec].properties[template].properties[spec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property]
```

**Expected behavior**
CPAO should install correctly.

**Kubernetes Details (`kubectl version`):**
K8s `v1.18.x`

**Custom Pod Autoscaler Operator Details:**
`v1.0.2

**Additional context**
This is blocked until operator-framework/operator-sdk#3235 is resolved. If this is required urgently a temporary fix could be added to manually modify the CRD YAML, to make the protocol property required.

**WORKAROUND**
A workaround for this is to install using the following steps:

1. Download the helm release to install:

```bash
VERSION=v1.0.2
wget https://github.com/jthomperoo/custom-pod-autoscaler-operator/releases/download/${VERSION}/custom-pod-autoscaler-operator-${VERSION}.tgz
```

2. Untar the helm package and then delete it:

```bash
tar -xzf custom-pod-autoscaler-operator-${VERSION}.tgz && rm custom-pod-autoscaler-operator-${VERSION}.tgz
```

3. Edit `custom-pod-autoscaler-operator/templates/crd/custompodautoscaler.com_custompodautoscalers.yaml`, replacing:

```yaml
                              required:
                              - containerPort
                              type: object
                            type: array
                            x-kubernetes-list-map-keys:
                            - containerPort
                            - protocol
```

with:

```yaml
                              required:
                              - containerPort
                              - protocol
                              type: object
                            type: array
                            x-kubernetes-list-map-keys:
                            - containerPort
                            - protocol
```

There should be two instances of this to replace.

4. Tar the edited helm package back up:

```bash
tar -czvf custom-pod-autoscaler-operator-${VERSION}.tgz custom-pod-autoscaler-operator/
```

5. Install using the modified helm package:

```bash
HELM_CHART=custom-pod-autoscaler-operator
helm install ${HELM_CHART} custom-pod-autoscaler-operator-${VERSION}.tgz
```

The CPAO should install successfully, check with `helm ls` and `kubectl get pods`.

**Comment by uptickmetachu** [comment]
According to this issue: https://github.com/operator-framework/operator-sdk/issues/3235
This has now been fixed and unblocked.

**Comment by jthomperoo** [comment]
This has now been fixed, it is available in version `v1.0.3` of the Custom Pod Autoscaler Operator, thanks for pointing out this had been unblocked!
